src/floris/simulation/flow_field.py

```diff
@@ -89,3 +102,18 @@ def finalize(self, unsorted_indices):
         self.u = np.take_along_axis(self.u, unsorted_indices, axis=2)
         self.v = np.take_along_axis(self.v, unsorted_indices, axis=2)
         self.w = np.take_along_axis(self.w, unsorted_indices, axis=2)
+
+    def calculate_speed_ups(self, het_map, x, y, z=None):
```

Comment:
A couple of questions about this function: 1) How often is called?  2) It has a list comprehension, is that fast enough?

Reply:
Good question, so this function gets called once per solve execution, at the beginning when the flow field is initialized based on the specified grid. And for a "full load" calculation of 100 turbines, 72 wind directions, and 20 wind speeds, the `calculate_speed_ups` function call takes approx. 0.01 seconds. So in my opinion, it is fast enough.